File: src/main/java/com/vyperplugin/VyperFileType.java
package com.vyperplugin;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class VyperFileType extends LanguageFileType {

    public static final VyperFileType INSTANCE = new VyperFileType();

    private VyperFileType(){
        super(Vyper.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Vyper file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Vyper language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "vy";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return VyperIcons.FILE;
    }
}

File: src/main/kotlin/com/vyperplugin/VyperFileType.kt
package com.vyperplugin

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object VyperFileType  : LanguageFileType(VyperLanguage) {

    override fun getName(): String {
        return "Vyper file"
    }

    override fun getDescription(): String {
        return "Vyper language file"
    }

    override fun getDefaultExtension(): String {
        return "vy"
    }

    override fun getIcon(): Icon? {
        return VyperIcons.FILE
    }
}
